docs/user-guide.rst

```diff
@@ -0,0 +1,391 @@
+User Guide
+==========
+
+.. currentmodule:: urllib3
+
+Making requests
+---------------
+
+First things first, import the Urllib3 module::
+
+    >>> import urllib3
+
+You'll need a :class:`~poolmanager.PoolManager` instance to make requests.
+This object handles all of the details of connection pooling and thread safety
+so that you don't have to::
+
+    >>> http = urllib3.PoolManager()
+
+To make a request use :meth:`~poolmanager.PoolManager.request`::
+
+    >>> r = http.request('GET', 'http://httpbin.org/robots.txt')
+    >>> r.data
+    b'User-agent: *\nDisallow: /deny\n'
+
+``request()`` returns a :class:`~response.HTTPResponse` object, the
+:ref:`response_content` section explains how to handle various responses.
+
+You can use :meth:`~poolmanager.PoolManager.request` to make requests using any
+HTTP verb::
+
+    >>> r = http.request(
+    ...     'POST',
+    ...     'http://httpbin.org/post',
+    ...     fields={'hello: 'world'})
+
+The different types of requests you can send is covered in :ref:`request_data`.
+
+.. _response_content:
+
+Response content
+----------------
+
+The :class:`~response.HTTPResponse` object provides ``status``, ``data``, and
+``header`` attributes::
+
+    >>> r = http.request('GET', 'http://httpbin.org/ip')
+    >>> r.status
+    200
+    >>> r.data
+    b'{\n  "origin": "104.232.115.37"\n}\n'
+    >>> r.headers
+    HTTPHeaderDict({'Content-Length': '33', ...})
+
+JSON content
+~~~~~~~~~~~~
+
+The simpliest way to load JSON content is to read the whole response into
+memory::
+
+    >>> import json
+    >>> r = http.request('GET', 'http://httpbin.org/ip')
+    >>> json.loads(r.data.decode('utf-8'))
+    {'origin': '127.0.0.1'}
+
+However, this might not be the best way to handle larger responses. In these
+cases, you might want to stream the response::
+
+    >>> import codecs
+    >>> reader = codecs.getreader('utf-8')
+    >>> r = http.request(
+    ...     'GET',
+    ...     'http://httpbin.org/ip',
+    ...     preload_content=False)
+    >>> json.load(reader(r))
+    {'origin': '127.0.0.1'}
+    >>> r.release_conn()
```

Comment:
If the request is not fully read, then yes. We need the ContextManager thing to make this cleaner.